Define PACTA Params Schema

AlexAxthelm opened the pull request:
* Make use of the parameter inheritence from `pacta.workflow.utils:::parse_params()`.
  * Parameters to inherit are located as files in `inst/extdata/parameters` (accessible within R using `system.file(..., package = "workflow.pacta")`.
* Split system config out from portfolio parameters
  * Config is controlled via envvars, params are passed as JSON.
* Change mount points to container to `/mnt`


- [x] Depends on: https://github.com/RMI-PACTA/pacta.workflow.utils/pull/13
- [x] Depends on https://github.com/RMI-PACTA/pacta.portfolio.allocate/pull/49

Closes #62 
Closes #67 
Closes #66

AlexAxthelm committed:
Move pacta data to mount directly

rather than through hodlings date

AlexAxthelm committed:
Merge branch 'main' into feat/62-input-params-schema

AlexAxthelm committed:
Update path for 22Q4 test

AlexAxthelm committed:
Fix 23Q4 test data path

AlexAxthelm committed:
Move default params to system file

AlexAxthelm committed:
Use parameter inheritence

AlexAxthelm committed:
Update testing config for 23Q4

AlexAxthelm committed:
Update params for 2022Q4

AlexAxthelm committed:
Increment version number to 0.0.0.9003

AlexAxthelm committed:
Disentangle portfolio directory and filenames

AlexAxthelm committed:
Nicely format default params

AlexAxthelm committed:
remove directory paths from parameters

Closes #67

AlexAxthelm committed:
fixes for missing variables

AlexAxthelm committed:
Fix function name

AlexAxthelm committed:
Add checks for missing config values

AlexAxthelm committed:
Add config envvars to testing workflow

AlexAxthelm committed:
Fix: line continuations in test script

AlexAxthelm committed:
Fix output path

AlexAxthelm committed:
Change bind points for external FS to `/mnt`

Closes #66

AlexAxthelm committed:
Add portfolios Dir to Rbuildignore

AlexAxthelm committed:
Define default Mountpoints in Dockerfile

AlexAxthelm committed:
Update variable name in testing workflow

AlexAxthelm committed:
update README

AlexAxthelm committed:
Add initial Schema file

AlexAxthelm committed:
Remove `has_map` parameter

Always output map results as part of analysis, even if they are not
used.

AlexAxthelm committed:
Add `jsonvalidate` dependency

AlexAxthelm committed:
Add `libnode-dev` to support `jsonvalidate`

AlexAxthelm committed:
Add value validation to json schema

AlexAxthelm committed:
Use dev branch of `allocate` repo

AlexAxthelm committed:
Update calls and schema to use new allocate sigs

AlexAxthelm committed:
Update ID for full parameter schema

AlexAxthelm committed:
update required properties of full param set

AlexAxthelm committed:
Add validation for raw params (before inherit)

AlexAxthelm committed:
Add Test with full param definition, no inherit

AlexAxthelm commented:
@cjyetman @jdhoffa marking this as ready for review, with the caveat that I'm probably going to do some re-organization of the top level keys of the JSON params, based on what I see coming with [`workflow.pacta.report`](https://github.com/RMI-PACTA/workflow.pacta.report/issues/12). We've previously discussed having the JSON along the lines of 
```json
{
  "portfolio_files": ["1.csv", "2.csv"],
  "analysis": {
     "start_year": 2023,
     "equity_market_list": ["blah blah blah"]
   },
   "reporting": {
     "portfolio_name": "Foo Portfolio",
     "investor_name": "Bar Investor"
   }
}
```

which is not implemented here, but that can probably come as a separate PR

cjyetman commented on DESCRIPTION:
```diff
@@ -27,16 +27,17 @@ Roxygen: list(markdown = TRUE)
 Remotes:
-    RMI-PACTA/pacta.portfolio.allocate,
+    RMI-PACTA/pacta.portfolio.allocate@fix/48-remove-globalenv,
     RMI-PACTA/pacta.portfolio.audit,
     RMI-PACTA/pacta.portfolio.import,
     RMI-PACTA/pacta.portfolio.utils,
-    RMI-PACTA/pacta.workflow.utils
+    RMI-PACTA/pacta.workflow.utils@fix/integration-issues-with-worfklow.pacta
```
do you want the branch names merged to main?

AlexAxthelm replied:
I'd rather not. Those are the two PRs mentioned at the top.

AlexAxthelm replied:
resolved.

AlexAxthelm committed:
Use main branch for allocate

https://github.com/RMI-PACTA/pacta.portfolio.allocate/pull/49 is merged

jdhoffa commented on DESCRIPTION:
```diff
@@ -39,4 +40,4 @@ Remotes:
     RMI-PACTA/pacta.portfolio.import,
     RMI-PACTA/pacta.portfolio.utils,
-    RMI-PACTA/pacta.workflow.utils
+    RMI-PACTA/pacta.workflow.utils@fix/integration-issues-with-worfklow.pacta
```
I approved https://github.com/RMI-PACTA/pacta.workflow.utils/pull/13

AlexAxthelm replied:
reverted.

jdhoffa reviewed (changes requested):
Comment to revert changes to `DESCRIPTION` otherwise LGTM

AlexAxthelm committed:
Use main Branch of workflow utils

AlexAxthelm committed:
Adapt to signature change in function

from late revision in https://github.com/RMI-PACTA/pacta.portfolio.allocate/pull/49

Specifically https://github.com/RMI-PACTA/pacta.portfolio.allocate/pull/49/commits/0bdbac8a39de952f937ae2e6c9399e169cb509ef
